framer/Layer.coffee

```diff
@@ -209,6 +210,111 @@ class exports.Layer extends BaseClass
 			@_element.setAttribute "name", value
 
 	##############################################################
+	# Matrices
+
+	@define "matrix",
+		get: ->
+			if @force2d
+				return @_matrix2d
+			new Matrix()
+				.translate(@x, @y, @z)
+				.scale(@scale)
+				.scale(@scaleX, @scaleY, @scaleZ)
+				.skew(@skew)
+				.skewX(@skewX)
+				.skewY(@skewY)
+				.translate(0, 0, @originZ)
+				.rotate(@rotationX, 0, 0)
+				.rotate(0, @rotationY, 0)
+				.rotate(0, 0, @rotationZ)
+				.translate(0, 0, -@originZ)
+
+	@define "_matrix2d",
+		get: ->
+			new Matrix()
+				.translate(@x, @y)
+				.scale(@scale)
+				.skewX(@skew)
+				.skewY(@skew)
+				.rotate(0, 0, @rotationZ)
+
+	@define "transformMatrix",
+		get: ->
+			new Matrix()
+				.translate(@originX * @width, @originY * @height)
+				.multiply(@matrix)
+				.translate(-@originX * @width, -@originY * @height)
+
+	_perspectiveProjectionMatrix: (element) =>
+		p = element.perspective
+		m = new Matrix()
+		m.m34 = -1/p if p? and p isnt 0
+		m
+
+	_perspectiveMatrix: (element) =>
+		if !element
+			element = Screen
+		ox = element.perspectiveOriginX * element.width
+		oy = element.perspectiveOriginY * element.height
+		ppm = @_perspectiveProjectionMatrix(element)
+		new Matrix()
+			.translate(ox, oy)
+			.multiply(ppm)
+			.translate(-ox, -oy)
+
+	@define "matrix3d",
+		get: ->
+			ppm = @_perspectiveMatrix(@superLayer)
+			new Matrix()
+				.multiply(ppm)
+				.multiply(@transformMatrix)
+
+	_screenPoint: (point = {}, context = false) =>
+		point =
+			x: point.x || 0
+			y: point.y || 0
+			z: point.z || 0
+		point = @matrix3d.point(point)
+
+		for layer in @superLayers(context)
+			point.z = 0 if layer.flat
+			point = layer.matrix3d.point(point)
+			if !layer.superLayer
+				point.z = 0
+
+		point
+
+	_fromScreenPoint: (point = {}, context = false) =>
+		point =
```

Comment:
You can use `_.defaults` for this: `point = `_.defaults(point, {x:0, y:0, z:0})`